### DISCUSSION TITLE
No kids in the sample images of NSFW models

### DISCUSSION THREAD

**Discussion by Pdonor** [civitai/civitai#192] [Ideas]
I hope this is self-explanatory.

Or ... anything that has to do with kids should be far away from anything that has to do with adult content. The website looks really, really creepy as soon as you turn on NSFW, as it seems to alternate between NSFW model and something with a child.

**Comment by JustMaier** [comment]
This is absolutely our policy and this kind of content is against our terms of service. If you see this please report it. There are context menus on models, reviews, and comments with the ability to report any of them.

**Reply by cameronedward1995-del** [reply]
Justin, I would not like for you to have these updates and got me banned from your secrets right now!

**Comment by maxhulker** [comment]
@Ambaseman We'll be adding the ability to report individual images as well, this will help in our moderation efforts for narrowing down problematic content.

**Reply by gsgoldma** [reply]
why dont you guys just have an approval process for sample images?

**Reply by maxhulker** [reply]
That would be a really large number of images to have to approve each day, and we're a small team.

**Reply by Pdonor** [reply]
In that case, may I suggest that upon submitting an image, require people to check a box as to whether the image contains any character/person who is or appears to be under the age of 18, and another checkbox as to whether the image is NSFW or portrays adult-only content? Then any image that meets either criteria could be reviewed before going on the site.   Having a tag indicating that an image/embedding includes a minor could make it possible to find a way to ensure such images are not displayed alongside NSFW images.

**Reply by JustMaier** [reply]
We have a system like what you described in place for dealing with NSFW content depicting real people and it seems like typically people just don't check the boxes likely they should, so even if we added this, it's likely the same thing would happen here.

We do already have a few things in place that you may not have noticed:
1. We use AI to automatically detect whether an image is NSFW or not (using NSFW.js)
2. We have a blocklist of words and we scan prompts for those words. If the image is detected as NSFW and it includes any of the words from the blocklist, then we block them from uploading the image and submitting the content.

However, this only works if the image includes metadata and if they haven't tampered with the metadata.

Ideally, we could use an AI tool to automatically detect age in the same way we're detecting NSFW content and use that to flag something as needing review prior to it being visible on the site. I have seen some AI tools that do this, but none of them appear to be open source. If you or anyone else reading this knows of one, please let me know 👍

**Reply by ThereAreThoseWhoCallMeDave** [reply]
https://age.toolpie.com/
?

**Reply by JustMaier** [reply]
Yeah, I'm aware of this tool and others like it, but we need one that we can include in our image processing pipeline. Ideally, it'd be something that we could run client-side with Tensorflow.js so that we can get the uploaders attention before we even process the upload.

**Reply by khushil** [reply]
Perhaps something like https://github.com/smahesh29/Gender-and-Age-Detection - but to be honest this feels like a few spike sprints to test out the various options, not sure if your pipeline is open source or anything to get more hands-on help.

**Reply by WASasquatch** [reply]
> However, this only works if the image includes metadata and if they haven't tampered with the metadata.

That's a big one. I don't use metadata, and advise everyone not to, as it's not only abusing what PNG Tags (based on TIFF Tags) are for, which is authorship, copyright, and camera info for cataloging reference (not recreation). Additionally I've warned so many people that they were selling something that could be loaded right into WebUI and recreated, exactly, plus all the variations and intent of sale, without purchasing.

**My suggestion** is to have a hover button on images, similar to the prompt button, to easily flag a image for review by moderators. sub-menus, being out of sight, won't help as much as a button right there while you'd hovering over something or skimming.

**Reply by JustMaier** [reply]
Looks like [face-api.js includes age estimation](https://github.com/justadudewhohacks/face-api.js#models-age-and-gender-recognition), but it appears it's fairly inaccurate.

@khushil the whole project is open-source (this repo), so if you're interested in getting hands on let me know. I'd like to put this into the image processing pipeline that includes the automated NSFW detection.

**Reply by cameronedward1995-del** [reply]
Your updates got me responsible for this!

**Comment by Pdonor** [comment]
Look, the problem isn’t face detection. The problem is that the site is
crawling with pedophiles. Right now the word ‘Loli’ is being used as an
official, searchable trigger word on models that have been up for *days*:
https://civitai.com/models/7922/jibril-no-game-no-life-lora  "Verified: 2
days ago
SafeTensor  "’. Even when the browsing mode is set to  "safe" instead of
"nsfw" almost everything posted in "new" is either a girl with cleavage
showing or a girl in a schoolgirl uniform, Look, split the site into two.
Completely different URL for AdultCivitAI dot com or whatever. That site,
delete images or models of anyone that you’d card if she ordered a beer. On
the main vanilla site, act like you’re run by schoolmarms and don’t even
let people post bikini pictures. Make it a website that an art teacher can
show to students without the police being called. There. Just as much free
speech, easier to keep safe for people of all ages.

**Reply by chavinlo** [reply]
loli is mostly meant for anime though

**Reply by JustMaier** [reply]
We're working on something that I believe will address this. We wrote about it about a week ago in [a post on Reddit](https://www.reddit.com/r/civitai/comments/10t0xdu/heres_how_were_planning_on_improving_browsability/). This should help create a space for all types of content rather than having everything get crowded out by the swarm of anime waifu content.

**Reply by WASasquatch** [reply]
> fighting unexisting windmills now? I truly agree with every procedures to keep the site safe but let's be real: Hentai/Anime is a thing and loli content is kinda everyday normal routine in that area, **nothing to do with pedophiles**. Also a good 90% of CivitAI's users are on the site only for anime so your fight against that tag sounds like a nosense boomer complain to me (try watch some hentai then understand how the world really works, dude..)
> 
> An idea may be: keeping anime and else with separate tags (this may probably keep boomers and young people in separate areas too) but thats may be too much generic and wont work i think

Lol. Let's not pretend loli is something it isn't. Lolicon is well known to be about underage people (aka minors) engaging in sexual activities.

**Reply by chavinlo** [reply]
> > fighting unexisting windmills now? I truly agree with every procedures to keep the site safe but let's be real: Hentai/Anime is a thing and loli content is kinda everyday normal routine in that area, **nothing to do with pedophiles**. Also a good 90% of CivitAI's users are on the site only for anime so your fight against that tag sounds like a nosense boomer complain to me (try watch some hentai then understand how the world really works, dude..)
> > An idea may be: keeping anime and else with separate tags (this may probably keep boomers and young people in separate areas too) but thats may be too much generic and wont work i think
> 
> Lol. Let's not pretend loli is something it isn't. Lolicon is well known to be about underage people (aka minors) engaging in sexual activities.

*drawings
fixed 😊

**Reply by WASasquatch** [reply]
> > > fighting unexisting windmills now? I truly agree with every procedures to keep the site safe but let's be real: Hentai/Anime is a thing and loli content is kinda everyday normal routine in that area, **nothing to do with pedophiles**. Also a good 90% of CivitAI's users are on the site only for anime so your fight against that tag sounds like a nosense boomer complain to me (try watch some hentai then understand how the world really works, dude..)
> > > An idea may be: keeping anime and else with separate tags (this may probably keep boomers and young people in separate areas too) but thats may be too much generic and wont work i think
> > 
> > 
> > Lol. Let's not pretend loli is something it isn't. Lolicon is well known to be about underage people (aka minors) engaging in sexual activities.
> 
> *drawings fixed 😊

Right, cause when it's anime, or a "drawing" it's suddenly no longer a *depiction*. lol

**Reply by chavinlo** [reply]
> > > > fighting unexisting windmills now? I truly agree with every procedures to keep the site safe but let's be real: Hentai/Anime is a thing and loli content is kinda everyday normal routine in that area, **nothing to do with pedophiles**. Also a good 90% of CivitAI's users are on the site only for anime so your fight against that tag sounds like a nosense boomer complain to me (try watch some hentai then understand how the world really works, dude..)
> > > > An idea may be: keeping anime and else with separate tags (this may probably keep boomers and young people in separate areas too) but thats may be too much generic and wont work i think
> > > 
> > > 
> > > Lol. Let's not pretend loli is something it isn't. Lolicon is well known to be about underage people (aka minors) engaging in sexual activities.
> > 
> > 
> > *drawings fixed 😊
> 
> Right, cause when it's anime, or a "drawing" it's suddenly no longer a _depiction_. lol

https://www.justice.gov/criminal-ceos/citizens-guide-us-federal-law-child-pornography

>Images of child pornography are not protected under First Amendment rights, and are illegal contraband under federal law. Section 2256 of Title 18, United States Code, defines child pornography as any visual depiction of sexually explicit conduct involving a minor (someone under 18 years of age).  Visual depictions include photographs, videos, digital or computer generated images **indistinguishable from an actual minor**, and images created, adapted, or modified, but appear to depict an **identifiable, actual minor**.  Undeveloped film, undeveloped videotape, and electronically stored data that can be converted into a visual image of child pornography are also deemed illegal visual depictions under federal law.

>indistinguishable from an actual minor

>indistinguishable

And if you mention the PROTECT act

> Because the statute criminalized material that was neither child pornography involving actual children under Ferber nor obscenity under Miller, the Supreme Court concluded the CPPA sought to reach beyond obscene material, and the statute was struck down as overbroad and unconstitutional.

https://law.justia.com/cases/federal/district-courts/FSupp2/564/996/1407869/

**Reply by WASasquatch** [reply]
> > > > > fighting unexisting windmills now? I truly agree with every procedures to keep the site safe but let's be real: Hentai/Anime is a thing and loli content is kinda everyday normal routine in that area, **nothing to do with pedophiles**. Also a good 90% of CivitAI's users are on the site only for anime so your fight against that tag sounds like a nosense boomer complain to me (try watch some hentai then understand how the world really works, dude..)
> > > > > An idea may be: keeping anime and else with separate tags (this may probably keep boomers and young people in separate areas too) but thats may be too much generic and wont work i think
> > > > 
> > > > 
> > > > Lol. Let's not pretend loli is something it isn't. Lolicon is well known to be about underage people (aka minors) engaging in sexual activities.
> > > 
> > > 
> > > *drawings fixed 😊
> > 
> > 
> > Right, cause when it's anime, or a "drawing" it's suddenly no longer a _depiction_. lol
> 
> https://www.justice.gov/criminal-ceos/citizens-guide-us-federal-law-child-pornography
> 
> > Images of child pornography are not protected under First Amendment rights, and are illegal contraband under federal law. Section 2256 of Title 18, United States Code, defines child pornography as any visual depiction of sexually explicit conduct involving a minor (someone under 18 years of age).  Visual depictions include photographs, videos, digital or computer generated images **indistinguishable from an actual minor**, and images created, adapted, or modified, but appear to depict an **identifiable, actual minor**.  Undeveloped film, undeveloped videotape, and electronically stored data that can be converted into a visual image of child pornography are also deemed illegal visual depictions under federal law.
> 
> > indistinguishable from an actual minor
> 
> > indistinguishable
> 
> And if you mention the PROTECT act
> 
> > Because the statute criminalized material that was neither child pornography involving actual children under Ferber nor obscenity under Miller, the Supreme Court concluded the CPPA sought to reach beyond obscene material, and the statute was struck down as overbroad and unconstitutional.
> 
> https://law.justia.com/cases/federal/district-courts/FSupp2/564/996/1407869/

Falsifiable citation to serve your agenda is negligent.

>Defendant's argument that he has an unlimited constitutional right to possess obscene materials must fail. Free Speech Coalition made clear that banned material must meet either the Ferber or Miller standards. There is no dispute the images in this case do not involve real children, thus Ferber is inapplicable. Subsections 1466A(a) (1) and (b) (1) require the visual depictions be obscene and thus must satisfy the Miller standard. Because subsections 1466A(a) (2) and (b) (2) do not require the visual depictions be obscene, and those subsections of the statute restrict protected speech, the Court finds those sections are constitutionally infirm and cannot form the basis of a charge under the superseding indictment.

>The superseding indictment alleges sufficient facts to support counts charging violations of 18 U.S.C. § 1466A(a) (1) and (b) (1). Thus, the defects in subsections 1466A(a) (2) and (b) (2) do not necessitate dismissal of the superseding indictment in this case. The determination of what constitutes obscenity is a determination to be made by the trier of fact in this case. Accordingly, the Court finds the pending motion must be denied in part and granted in part. Defendant's Motion to Dismiss (Clerk's No. 45) is denied, with the exception of any prosecution under subsections 1466A(a) (2) and (b) (2) as to which the motion is granted.

>IT IS SO ORDERED.

The superseding indictment, or primary point of the case wasn't dismissed. Only part of the claims dismissed. (a) (1) and (b) (1) still stuck. 
https://www.law.cornell.edu/uscode/text/18/1466A

>(1)
>(A) depicts a minor engaging in [sexually explicit conduct] and
> (B) is obscene; or

And additional offense (of same case). 

The only thing dropped was about *real* depictions of a minor, but the depictions of a minor in general stuck (such as cartoon, painting, sculptor, etc). 

Furthermore, diffusion often does use real photography for training data, so that sort of indictment could stick in situations here at civitai. We already got people training a whole TI on underage Emma Watson and other people and trying to make up excuses about why it won't do them as an adult... 

I have been running anime websites and websites in general for over 25 years, there is a reason places like Danbooru, 4chan, etc, sever hop or TLD hop or host in counties lax in regulation... You either actively fight it, or you know what you are hosting and dodge like shady sites like danbooru

**Reply by lottepeisch** [reply]
> We're working on something that I believe will address this. We wrote about it about a week ago in [a post on Reddit](https://www.reddit.com/r/civitai/comments/10t0xdu/heres_how_were_planning_on_improving_browsability/). This should help create a space for all types of content rather than having everything get crowded out by the swarm of anime waifu content.

Is there an ETA on this rolling out? 

I'm accustomed to having to deal with the porn factor in any bleeding edge technology, but after months of having to sift through all the pedo/porn models to find what I want, I'm growing weary and visiting the site less and less. It's aggravating, and the NSFW/SFW tagging doesn't seem to work because people think upskirt shots and cleavage busting out of clothing is somehow SFW. 

Blocking users helps, blocking tags not-so-much since no one seems to know the difference between NSFW and SFW and a lot of people don't even use the tags too much, and you miss out on models that can do both sides well because blocking one tag blocks them all, and that doesn't even work entirely. I have anime blocked right now, and I still get about two new upskirt/busty/grossly-submissive models popping up several times a day.  

So yeah, any ETA on relieving this torture? Blocking 'Anime' gets a lot of it out, but it also blocks out models that can do Anime and other things, and it doesn't at all capture models put out by people who don't use the tagging system at all. Also, the front page makes the community look like a bunch of pedo pervs, and that's what journalists are seeing of us... not sure if you wanna be on the front lines of spreading that image. 

To be clear, as an end user, I do not want to see something and then have to report it... I don't want to see it at all. Blocking tags doesn't work, the NSFW filter doesn't work. HELP!

**Reply by WASasquatch** [reply]
> > We're working on something that I believe will address this. We wrote about it about a week ago in [a post on Reddit](https://www.reddit.com/r/civitai/comments/10t0xdu/heres_how_were_planning_on_improving_browsability/). This should help create a space for all types of content rather than having everything get crowded out by the swarm of anime waifu content.
> 
> Is there an ETA on this rolling out?
> 
> I'm accustomed to having to deal with the porn factor in any bleeding edge technology, but after months of having to sift through all the pedo/porn models to find what I want, I'm growing weary and visiting the site less and less. It's aggravating, and the NSFW/SFW tagging doesn't seem to work because people think upskirt shots and cleavage busting out of clothing is somehow SFW.
> 
> Blocking users helps, blocking tags not-so-much since no one seems to know the difference between NSFW and SFW and a lot of people don't even use the tags too much, and you miss out on models that can do both sides well because blocking one tag blocks them all, and that doesn't even work entirely. I have anime blocked right now, and I still get about two new upskirt/busty/grossly-submissive models popping up several times a day.
> 
> So yeah, any ETA on relieving this torture? Blocking 'Anime' gets a lot of it out, but it also blocks out models that can do Anime and other things, and it doesn't at all capture models put out by people who don't use the tagging system at all. Also, the front page makes the community look like a bunch of pedo pervs, and that's what journalists are seeing of us... not sure if you wanna be on the front lines of spreading that image.
> 
> To be clear, as an end user, I do not want to see something and then have to report it... I don't want to see it at all. Blocking tags doesn't work, the NSFW filter doesn't work. HELP!

I agree there a lot. Maybe some sort of separation between adult content (18+) and general NSFW (gore, risque, etc). Even further would be cool like, Porn, versus Adult (Mature themes, gore, occasional nudity), and NSFW which covers them all plus risque.

**Reply by lottepeisch** [reply]
> I agree there a lot. Maybe some sort of separation between adult content (18+) and general NSFW (gore, risque, etc). Even further would be cool like, Porn, versus Adult (Mature themes, gore, occasional nudity), and NSFW which covers them all plus risque.

Yeah, that would be great. The rough part is the fact that some models can do both, and it seems like a lot of folks choose their sample images based on the idea that if it has tits, it gets hits. DreamShaper, for example, does some amazing things that aren't anime or hot chicks or anything close to NSFW, but you wouldn't know that by the sample images. I spend an inordinate amount of time having to sift through comments and hope someone posted things other than hot chicks so I can see the whole breadth of the model and not just the parts the uploader knows will get them hits. 

I don't have the answer to the 'how', just the use case wish to see the full breadth of what a model can do rather than just the pervy stuff the creator knows will guarantee downloads and a place on the leaderboard, which is almost entirely NSFW crap. 

And let's be clear on what Not Safe For Work means, gang... MOST of what I've seen marked SFW would be unacceptable in nearly every workplace imaginable except porn shops and sleazy mechanics in the backwoods. If you'd get sent home for wearing it at work, it's not safe for work!

**Reply by JustMaier** [reply]
> Is there an ETA on this rolling out?

End of this week. We also hope to have the ai age detection with automatic flagging for review ready by the end of the week as well...

**Reply by lottepeisch** [reply]
> > Is there an ETA on this rolling out?
> 
> End of this week. We also hope to have the ai age detection with automatic flagging for review ready by the end of the week as well...

Awesome! Thank you for your work!

**Comment by zzz6519003** [comment]
sample image?

**Comment by cameronedward1995-del** [comment]
I would not have this new update for this!
